From: jbeulich@suse.com
Subject: netfront: linearize SKBs requiring too many slots
Patch-mainline: Never, SUSE-Xen specific
References: bsc#991247

This will reduce the number of slots required, allowing to avoid the
dropping of such SKBs.

--- a/drivers/xen/netfront/netfront.c
+++ b/drivers/xen/netfront/netfront.c
@@ -1007,9 +1007,10 @@ static int network_start_xmit(struct sk_
 
 	slots = PFN_UP(offset + len) + xennet_count_skb_frag_slots(skb);
 	if (unlikely(slots > MAX_SKB_FRAGS + 1)) {
-		net_alert_ratelimited("xennet: skb rides the rocket: %u slots\n",
-				      slots);
-		goto drop;
+		net_dbg_ratelimited("xennet: skb rides the rocket: %u slots, %u bytes\n",
+				    slots, skb->len);
+		if (skb_linearize(skb))
+			goto drop;
 	}
 
 	spin_lock_irqsave(&np->tx_lock, flags);
